### PR TITLE
tests: covers macro coverage.

### PR DESCRIPTION
**Pull request by jcchavezs**
> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Make sure that you've checked the boxes below before you submit PR:**

- [ ] My code includes positive and negative tests.
- [ ] I have an appropriate description with correct grammar.
- [ ] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v3sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).
- [ ] My code is properly linted and passes pre-commit tests.

Thanks for your contribution :heart:

### CODE REVIEW

File: macro/macro.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ type Macro interface {
 }
 
 func NewMacro(data string) (Macro, error) {
+	// TODO(jcchavezs): shall we fail if data is empty?
```
Comment (jcchavezs) [inline macro/macro.go:21]: I think so, invoking a macro with no data is mainly a programming mistake? e.g. using passing a `msg` with no text.

Comment (jcchavezs) [inline macro/macro.go:21]: I think so, invoking a macro with no data is mainly a programming mistake? e.g. using passing a `msg` with no text.

File: macro/macro.go [hunk 2]
```diff
@@ -101,14 +107,16 @@ func (m *macro) compile(input string) error {
...
 			continue
 		}
-		if ismacro {
+
+		if isMacro {
 			if c == '}' {
 				// we close a macro
-				ismacro = false
+				isMacro = false
+				// TODO(jcchavezs): can key be empty? e.g. %{var}
```
Comment (jcchavezs) [inline macro/macro.go:119]: Ping @airween @fzipi @jptosso

File: macro/macro.go [hunk 3]
```diff
@@ -146,6 +164,7 @@ func (m *macro) String() string {
 }
 
 // IsExpandable return true if there are macro expanadable tokens
+// TODO(jcchavezs): this is used only in a commented out section
```
Comment (jcchavezs) [inline macro/macro.go:167]: cc @anuraaga

File: macro/macro.go (hunk 2 above)
Comment (jptosso) [reply macro/macro.go:119]: We should test in modsec

Comment (jptosso) [reply macro/macro.go:119]: Syntax error unless it's a single variable like request_uri

**Pull request merged**